catalogue/webapp/pages/work.js

```diff
@@ -19,6 +161,16 @@ const WorkPage = ({ work }: Props) => {
   );
   const iiifInfoUrl = iiifImageLocation && iiifImageLocation.url;
 
+  const sierraId = (work.identifiers.find(identifier =>
```

Comment:
What are the arguments for/against doing the data manipulation here, rather than in `getInitialProps` and passing the Work in pre-massaged?

Reply:
It's display logic, so feels like it should be in the display.

The fact we never use the data munging anywhere means that it's separated from where it's actually used, in my head leaving me having to make a cognitive leap as to "Why has this been abstracted?".

Is the concern about neatness? Or something else?

Reply:
No concern – merely a question.

Reply:
It's an interesting one, I personally find it hard to focus on a thing when I am bouncing around between the method and the use of that method.

Again, personally, when refactoring or changing a method that has been abstracted, I feel like I have to make it general, which can be a great challenge, but sometimes impractical and can tie you in knots.

These are personal styles though, @gestchild what do you think?